ext/linux/osq_lock.h

```diff
@@ -228,14 +290,14 @@ bool osq_lock(uint64_t *osq, unsigned long cpu_number)
      */
 
     while (!READ_ONCE(node->locked)) {
+        /* TODO: Need to emulate kernel rescheduling */
         /*
          * If we need to reschedule bail... so we can block.
          * Use vcpu_is_preempted() to avoid waiting for a preempted
-         * lock holder:
+         * lock holder.
          */
-        /* TODO: How to emulate rescheduling? */
         //if (need_resched() || vcpu_is_preempted(node_to_cpu(node->prev)))
-        if (++back_off > unqueue_retry)
+        if (++back_off > unqueue_retry) /* default MAX_UNQUEUE_RETRY */
```

Comment:
Need to place comment documentation here to explain what we are doing differently from the kernel as well.